lms/djangoapps/mobile_api/users/tests.py

```diff
@@ -1068,3 +1069,137 @@ def test_discussion_tab_url(self, discussion_tab_enabled):
+        self.login()
+        courses = [CourseFactory.create(org="edx", mobile_available=False) for _ in range(3)]
+        for course in courses:
+            self.enroll(course.id)
+
+        response = self.api_response(api_version=API_V1)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertListEqual(response.data, [])
+
+    def test_no_enrollments(self) -> None:
+        self.login()
+        for _ in range(3):
+            CourseFactory.create(org="edx", mobile_available=True)
+
+        response = self.api_response(api_version=API_V1)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertListEqual(response.data, [])
+
+    def test_user_have_only_active_enrollments_and_no_completions(self) -> None:
+        self.login()
+        courses = [CourseFactory.create(org="edx", mobile_available=True) for _ in range(3)]
+        for course in courses:
+            self.enroll(course.id)
+
+        response = self.api_response(api_version=API_V1)
+
+        expected_response = [
+            {'course_id': str(courses[0].course_id), 'course_name': courses[0].display_name, 'is_active': True},
+            {'course_id': str(courses[1].course_id), 'course_name': courses[1].display_name, 'is_active': True},
+            {'course_id': str(courses[2].course_id), 'course_name': courses[2].display_name, 'is_active': True},
+        ]
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertListEqual(response.data, expected_response)
+
+    def test_user_have_active_and_inactive_enrollments_and_no_completions(self) -> None:
+        self.login()
+        courses = [CourseFactory.create(org="edx", mobile_available=True) for _ in range(3)]
+        for course in courses:
+            self.enroll(course.id)
+        old_course = CourseFactory.create(org="edx", mobile_available=True)
+        self.enroll(old_course.id)
+        old_enrollment = CourseEnrollment.objects.filter(user=self.user, course=old_course.course_id).first()
+        old_enrollment.created = datetime.datetime.now(pytz.UTC) - datetime.timedelta(days=31)
+        old_enrollment.save()
+
+        response = self.api_response(api_version=API_V1)
+
+        expected_response = [
+            {'course_id': str(courses[0].course_id), 'course_name': courses[0].display_name, 'is_active': True},
+            {'course_id': str(courses[1].course_id), 'course_name': courses[1].display_name, 'is_active': True},
+            {'course_id': str(courses[2].course_id), 'course_name': courses[2].display_name, 'is_active': True},
+            {'course_id': str(old_course.course_id), 'course_name': old_course.display_name, 'is_active': False}
+        ]
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertListEqual(response.data, expected_response)
+
+    @ddt.data(
+        (27, True),
+        (28, True),
+        (29, True),
+        (31, False),
+        (32, False),
```

Comment:
[info] I think it is enough to have before border value, after border value and border value itself. Maybe we need to add `30` also. It is called Boundary Testing principle

Reply:
But no action needed here

Reply:
I agree, to have before border value, after border value it would be enough. 

However, it is possible that we would get a problem if we used the 30 day limit and ran the tests at approximately 00:00. So I removed this value. But this can also be easily handled using a decorator `@freeze_time('2018-01-01')`.

What do you think? Do I need to refactor these tests? It's pretty simple

Reply:
no, it is alright